x-pack/plugins/saved_objects_tagging/public/components/base/tag_badge.tsx

```diff
@@ -7,18 +7,36 @@
 
 import React, { FC } from 'react';
 import { EuiBadge } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
 import { Tag, TagAttributes } from '../../../common/types';
 
 export interface TagBadgeProps {
   tag: Tag | TagAttributes;
+  onClick?: (tag: Tag | TagAttributes) => void;
```

Comment:
For consistency I think we should be using `TagWithOptionalId` here (given it's kinda the exact same as `Tag | TagAttributes`). Can change the type for the `tag` property on previous line too. 

Same in `x-pack/plugins/saved_objects_tagging/public/components/base/tag_list.tsx`

Reply:
Done in https://github.com/elastic/kibana/pull/140947/commits/5aaf223f268a8db596534494fb6eef3c1c6fe329